### PR TITLE
Retry when yarn install fails

### PR DESCRIPTION
**Pull request by jez**
### Test plan

See included automated tests.

### CODE REVIEW

File: .buildkite/test-vscode-extension.sh [hunk 1]
```diff
@@ -10,7 +10,33 @@ apt-get install -y libxshmfence-dev libnss3-dev libatk1.0-0 libatk-bridge2.0-0 \
...
+
+  local attempt=0
+  while true; do
+    attempt=$(( attempt + 1 ))
+    echo "Attempt $attempt"
+    if "$@"; then
+      return 0
+    fi
+
+    if (( attempt >= attempts )); then
```
Comment (froydnj) [inline .buildkite/test-vscode-extension.sh:26]: Does this need a `$` prefix, or does the magic of the if make this get evaluated as an arithmetic expression?

**Review by froydnj — APPROVED** [review]
Looks good, just one question.

File: .buildkite/test-vscode-extension.sh (hunk 1 above)
Comment (jez) [reply .buildkite/test-vscode-extension.sh:26]: It works as is (I copied this code directly from test-static-sanitized.sh)

Comment (jez) [reply .buildkite/test-vscode-extension.sh:26]: I think that in particular the magic of the `(( ... ))` not the `if` makes it work.

Comment (jez) [reply .buildkite/test-vscode-extension.sh:26]: Also, writing `$((...))` triggers this shellcheck warning:

https://www.shellcheck.net/wiki/SC2084

Comment (froydnj) [reply .buildkite/test-vscode-extension.sh:26]: Ah, I see, I was not aware of the difference between [arithmetic expansion](https://www.gnu.org/software/bash/manual/html_node/Arithmetic-Expansion.html#Arithmetic-Expansion) and plain [shell arithmetic](https://www.gnu.org/software/bash/manual/html_node/Shell-Arithmetic.html).  TIL.

**Pull request merged**